cuda-ssh/Dockerfile

```diff
@@ -0,0 +1,53 @@
+ARG BASE_IMAGE="ghcr.io/coreweave/ml-containers/torch:bb02bee-base-cuda11.8.0-torch2.0.0-vision0.15.1-audio2.0.1"
+
+FROM ${BASE_IMAGE}
+
+RUN apt-get -qq update && \
+    DEBIAN_FRONTEND=noninteractive \
+      apt-get -qq install --no-install-recommends -y \
+      # Critical packages:
+      ssh ca-certificates tini bash \
+      # Helpful packages:
+      libncurses5 curl wget sudo htop git rsync locales \
+      tmux unzip nano vim apt-utils iputils-ping && \
+    apt-get clean && \
+    # SSH passes the client's LANG and LC_* environment variables by default.
+    # However, the only pre-installed locales on most container images are
+    # C, C.UTF-8, and POSIX. This adds the en_US.UTF-8 locale as well,
+    # and leaves locale-gen available to install others.
+    locale-gen en_US.UTF-8 && \
+    # Wipe the server-side SSH keys on the container image level
+    # to prevent leaking the private host keys, which could
+    # potentially allow impersonation of the SSH server by an attacker.
+    rm /etc/ssh/ssh_host_*
+
+# Since there are no host keys, the SSH server
+# MUST be configured at runtime by running:
+#   dpkg-reconfigure openssh-server
+# (Or by adding custom host key files to /etc/ssh/) before launching it with:
+#   service ssh start
+# Or (blocking):
+#   service ssh start -D
+
+RUN \
+    # Configure the privilege separation directory for sshd
+    # See here for details: https://github.com/openssh/openssh-portable/blob/master/README.privsep
+    install -d --mode=0755 --owner=0 --group=0 /var/run/sshd && \
+    # Configure an empty authorized keys file with correct permissions
+    install -d --mode=0700 --owner=0 --group=0 /root/.ssh && \
+    install --mode=600 --owner=0 --group=0 /dev/null /root/.ssh/authorized_keys && \
+    # Allow only public key authentication
+    install --mode=600 --owner=0 --group=0 /dev/null /etc/ssh/sshd_config.d/10-key-auth.conf && \
+    echo "PasswordAuthentication no" >> /etc/ssh/sshd_config.d/10-key-auth.conf && \
```

Comment:
Just curious if we make this a requirement to only use SSH Public Key or give them the option to use password?

Reply:
I was thinking we would make public key authentication the only option. If we want to also allow password authentication we could add an alternative configuration file called `11-password-auth.conf` and allow deleting `10-key-auth.conf` at the time of deployment, but the public key authentication setup is pretty straightforward, so I'd lean towards just that.

Reply:
I advocate no passwords, and nor should we encourage our customers to use passwords.